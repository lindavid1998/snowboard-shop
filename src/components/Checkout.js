import React from 'react';
import CartItem from './CartItem';
import { Link } from 'react-router-dom';

export default function Checkout(props) {
	const { cart, removeFromCart } = props;

	const deliveryInfo = (
		<fieldset>
			<legend>Delivery</legend>
			<div className='form-row'>
				<label htmlFor='name'>Name</label>
				<input type='text' id='name' name='name' />
			</div>
			<div className='form-row'>
				<label htmlFor='name'>Address</label>
				<input type='text' id='address' name='address' />
			</div>
			<div className='form-row'>
				<label htmlFor='name'>City</label>
				<input type='text' id='city' name='city' />
			</div>
			<div className='form-row'>
				<label htmlFor='name'>State</label>
				<input type='text' id='state' name='state' />
			</div>
			<div className='form-row'>
				<label htmlFor='name'>ZIP Code</label>
				<input type='text' id='zip' name='zip' />
			</div>
		</fieldset>
	);

	const paymentInfo = (
		<fieldset>
			<legend>Payment</legend>
			<div className='form-row'>
				<label htmlFor='name'>Name on card</label>
				<input type='text' id='name' name='name' />
			</div>
			<div className='form-row'>
				<label htmlFor='name'>Card number</label>
				<input type='text' id='card-number' name='card-number' />
			</div>
			<div className='form-row'>
				<label htmlFor='name'>Expiration date</label>
				<input type='text' id='exp-date' name='exp-date' />
			</div>
		</fieldset>
	);

	const items = (
		<fieldset>
			<legend>Cart</legend>
			{cart.map((item) => (
				<CartItem
					item={item}
					key={item.id}
					className='cart-item checkout'
					removeFromCart={removeFromCart}
				/>
			))}
		</fieldset>
  );
  
  const totalPrice = cart.reduce((acc, cur) => acc + cur.price, 0)

  const orderSummary = (
		<div className='order-summary'>
			<div className='total'>Total: {`$${totalPrice}`}</div>
			<Link to='/confirmation'>
				<input type='submit' value='Place Order' />
			</Link>
		</div>
	);

	return (
		<div className='Checkout'>
			<form action=''>
				{deliveryInfo}
				{paymentInfo}
				{items}
        {orderSummary}
			</form>
		</div>
	);
}
