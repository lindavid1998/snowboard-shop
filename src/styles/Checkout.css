.checkout-form {
  padding: 10px 40px 50px 40px;
  width: fit-content;
  display: grid;
  grid-template-columns: auto;
  grid-template-rows: auto;
  gap: 30px;
}

.checkout-cart-items {
  grid-area: 1 / 2 / -1 / -1;
  overflow: auto;
}

.checkout-info {
  display: flex;
  flex-direction: column;
  gap: 30px;
}

.checkout-form fieldset {
	border: 0;
  padding: 0;
}

.checkout-form legend {
	font-weight: bold;
	font-size: 1.2rem;
  padding: 0;
  margin-bottom: 10px;
}

.checkout-form input {
	background-color: #2d2d2d;
	border: 0;
	border: 1px solid #2d2d2d;
	padding: 10px;
	width: 250px;
	color: white;
  transition: 0.25s all ease-in-out;
}

input:focus {
	border: 1px solid white;
	outline: none;
}

.form-row label {
	display: block;
	margin-bottom: 3px;
  font-size: 0.8rem;
}

.form-rows {
	display: grid;
	grid-template-columns: repeat(2, 300px);
	gap: 12px;
}

.checkout-cart-items {
  display: flex;
  gap: 20px;
  flex-direction: column;
}

.order-summary {
  display: flex;
  align-items: flex-end;
  justify-content: space-between;
}

.total {
  font-size: 1.2rem;
  font-weight: bold;
}

.btn-place-order {
  position: relative;
  font-size: 1.2rem;
  padding: 0.5rem;
  right: 30%;
  background-color: rgb(0, 136, 16);
}

.btn-place-order:hover {
  background-color: rgb(0, 159, 19);
  color: white;
  cursor: pointer;
}