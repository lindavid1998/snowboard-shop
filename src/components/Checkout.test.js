import { render, screen } from '@testing-library/react';
import Checkout from './Checkout';
import { BrowserRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { act } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';

describe('Checkout', () => {
	test('renders cart items and payment form', () => {
		const mockCart = [
			{
				id: 0,
				name: 'Season Kin Snowboard 2023',
				price: 349.3,
				category: 'boards',
				qty: 1,
			},
			{
				id: 1,
				name: 'GNU Young Money C2E Snowboard - Big Boys 2023',
				price: 279.99,
				category: 'boards',
				qty: 2,
			},
		];
		const checkout = renderer
			.create(
				<BrowserRouter>
					<Checkout cart={mockCart} />
				</BrowserRouter>
			)
			.toJSON();
		expect(checkout).toMatchSnapshot();
	});

	test('triggers form submission when order is placed', async () => {
		const mockFn = jest.fn(e => e.preventDefault());
		const user = userEvent.setup();
		render(<Checkout submitOrder={mockFn} cart={[]} />);

		await act(async () => {
			await user.click(screen.getByText(/place order/i));
		});

		expect(mockFn).toHaveBeenCalledTimes(1);
	});

});
