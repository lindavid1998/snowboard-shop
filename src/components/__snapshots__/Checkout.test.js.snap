// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout renders cart items and payment form 1`] = `
<div
  className="Checkout"
>
  <form
    action=""
  >
    <fieldset>
      <legend>
        Delivery
      </legend>
      <div
        className="form-row"
      >
        <label
          htmlFor="name"
        >
          Name
        </label>
        <input
          id="name"
          name="name"
          type="text"
        />
      </div>
      <div
        className="form-row"
      >
        <label
          htmlFor="name"
        >
          Address
        </label>
        <input
          id="address"
          name="address"
          type="text"
        />
      </div>
      <div
        className="form-row"
      >
        <label
          htmlFor="name"
        >
          City
        </label>
        <input
          id="city"
          name="city"
          type="text"
        />
      </div>
      <div
        className="form-row"
      >
        <label
          htmlFor="name"
        >
          State
        </label>
        <input
          id="state"
          name="state"
          type="text"
        />
      </div>
      <div
        className="form-row"
      >
        <label
          htmlFor="name"
        >
          ZIP Code
        </label>
        <input
          id="zip"
          name="zip"
          type="text"
        />
      </div>
    </fieldset>
    <fieldset>
      <legend>
        Payment
      </legend>
      <div
        className="form-row"
      >
        <label
          htmlFor="name"
        >
          Name on card
        </label>
        <input
          id="name"
          name="name"
          type="text"
        />
      </div>
      <div
        className="form-row"
      >
        <label
          htmlFor="name"
        >
          Card number
        </label>
        <input
          id="card-number"
          name="card-number"
          type="text"
        />
      </div>
      <div
        className="form-row"
      >
        <label
          htmlFor="name"
        >
          Expiration date
        </label>
        <input
          id="exp-date"
          name="exp-date"
          type="text"
        />
      </div>
    </fieldset>
    <fieldset>
      <legend>
        Cart
      </legend>
      <li
        className="cart-item checkout"
        data-testid="cart-item"
        id={0}
      >
        <div
          className="item-name"
        >
          Season Kin Snowboard 2023
        </div>
        <div
          className="item-price"
        >
          $
          349.3
        </div>
        <button
          onClick={[Function]}
        >
          Remove
        </button>
      </li>
      <li
        className="cart-item checkout"
        data-testid="cart-item"
        id={1}
      >
        <div
          className="item-name"
        >
          GNU Young Money C2E Snowboard - Big Boys 2023
        </div>
        <div
          className="item-price"
        >
          $
          279.99
        </div>
        <button
          onClick={[Function]}
        >
          Remove
        </button>
      </li>
    </fieldset>
    <div
      className="order-summary"
    >
      <div
        className="total"
      >
        Total: 
        $629.29
      </div>
      <a
        href="/confirmation"
        onClick={[Function]}
      >
        <input
          type="submit"
          value="Place Order"
        />
      </a>
    </div>
  </form>
</div>
`;
